exporter.go

```diff
@@ -819,6 +856,23 @@ func (e *Exporter) extractTile38Metrics(ch chan<- prometheus.Metric, c redis.Con
 	}
 }
 
+func (e *Exporter) extractConnectedClientMetrics(ch chan<- prometheus.Metric, c redis.Conn) {
+	if reply, err := redis.String(doRedisCmd(c, "CLIENT", "LIST")); err == nil {
+		clients := strings.Split(reply, "\n")
+		
+		var clientsMap = map[string]int{}
```

Comment:
Why still aggregate in the exporter now that you have all the pieces of information about a client? 

You should submit the metric with all the labels so the loop in 870-872 is no longer needed and you can call `registerConstMetricGauge()` right away in 866.

Reply:
@oliver006 but I need a count of how many clients share names and such?
How would you do this?

Reply:
Once you have all the client details exported you can then get the aggregate in the Prometheus server by using PromQL and do something like `sum(connected_clients_details) by (name)`

Reply:
Also what metric, I still don't see it

Reply:
The metric would be `connected_clients_details` and the labels would be as defined by you in line 309 of this file.

Reply:
It would look something like

`connected_clients_details{host="10.x.x.x" port="4321" flags="aC"} 1`.

Reply:
this function: registerConstMetricGauge takes a float64 value so I still don't know what I'm passing to the function I just don't see it

Reply:
I see what you meant and I think I got it

Reply:
Yes, this looks a lot better! I'll give this another round of review later today.